src/commands/deployWorkspaceProject/internal/deployWorkspaceProjectInternal.ts

```diff
@@ -100,145 +108,83 @@ export async function deployWorkspaceProjectInternal(
     };
 
     const promptSteps: AzureWizardPromptStep<DeployWorkspaceProjectInternalContext>[] = [
-        new DeployWorkspaceProjectConfirmStep(!!options.suppressConfirmation),
-        new SharedResourcesNameStep(),
-        new AppResourcesNameStep(!!options.suppressContainerAppCreation)
+        new RootFolderStep(),
     ];
+    const executeSteps: AzureWizardExecuteStep<DeployWorkspaceProjectInternalContext>[] = [];
 
-    const executeSteps: AzureWizardExecuteStep<DeployWorkspaceProjectInternalContext>[] = [
-        new DeployWorkspaceProjectSaveSettingsStep()
-    ];
+    LocationListStep.addProviderForFiltering(context, managedEnvironmentProvider, managedEnvironmentResourceType);
+    LocationListStep.addProviderForFiltering(context, logAnalyticsProvider, logAnalyticsResourceType);
+    LocationListStep.addProviderForFiltering(context, registryProvider, registryResourceType);
+    LocationListStep.addProviderForFiltering(context, containerAppProvider, containerAppResourceType);
 
-    // Resource group
-    if (wizardContext.resourceGroup) {
-        wizardContext.telemetry.properties.existingResourceGroup = 'true';
-
-        const resourceGroupName: string = nonNullValueAndProp(wizardContext.resourceGroup, 'name');
+    if (!options.advancedCreate) {
+        // Basic
+        if (!wizardContext.resourceGroup) {
+            executeSteps.push(new ResourceGroupCreateStep());
+        }
+        if (!wizardContext.managedEnvironment) {
+            executeSteps.push(
+                new LogAnalyticsCreateStep(),
+                new ManagedEnvironmentCreateStep(),
+            );
+        }
+        if (!wizardContext.registry) {
+            executeSteps.push(new RegistryCreateStep());
+        }
+        if (!wizardContext.containerApp && !options.suppressContainerAppCreation) {
+            executeSteps.push(new ContainerAppCreateStep());
+        }
 
-        prependOrInsertAfterLastInfoChild(wizardContext,
-            new ActivityChildItem({
-                label: localize('useResourceGroup', 'Use resource group "{0}"', resourceGroupName),
-                activityType: ActivityChildType.Info,
-                contextValue: activityInfoContext,
-                iconPath: activityInfoIcon
-            })
+        promptSteps.push(
+            new SharedResourcesNameStep(),
+            new AppResourcesNameStep(!!options.suppressContainerAppCreation),
         );
-
-        await LocationListStep.setLocation(wizardContext, wizardContext.resourceGroup.location);
-        ext.outputChannel.appendLog(localize('usingResourceGroup', 'Using resource group "{0}".', resourceGroupName));
     } else {
-        wizardContext.telemetry.properties.existingResourceGroup = 'false';
-        executeSteps.push(new ResourceGroupCreateStep());
-    }
-
-    // Managed environment
-    if (wizardContext.managedEnvironment) {
```

Comment:
All of the starting resource displaying is now handled by `StartingResourcesLogStep` which allows me to remove a ton of boilerplate code from this file.  Now it's really only only a matter of pushing in the right basic vs. advanced steps.